ipaserver/plugins/idviews.py

```diff
@@ -715,6 +715,39 @@ def remove_ipaobject_overrides(ldap, api, dn):
             ldap.delete_entry(entry)
 
 
+# Extended user validator that allows to resolve users from trusted domains
+@member_validator(membertype='user')
+def validate_external_user(ldap, dn, keys, options, value):
+    trusted_objects = options.get('trusted_objects', [])
+    param = api.Object['user'].primary_key
+    uid = param(value)
+
+    try:
+        param.validate(uid)
+    except errors.ValidationError as e:
+        membertype = 'user'
+        try:
+            id = resolve_object_to_anchor(ldap, 'user', uid, False)
+            if id.startswith(SID_ANCHOR_PREFIX):
+                trusted_objects.append(uid)
+                options['trusted_objects'] = trusted_objects
+                return
+        except errors.NotFound:
+            membertype = 'group'
+            param = api.Object['group'].primary_key
+            group = param(value)
+            try:
+                id = resolve_object_to_anchor(ldap, 'group', group, False)
+                if id.startswith(SID_ANCHOR_PREFIX):
+                    group = '%:' + value
+                    trusted_objects.append(group)
+                    options['trusted_objects'] = trusted_objects
+                    return
+            except errors.NotFound:
+                pass
+            raise errors.ValidationError(name=membertype, error=e.error)
```

Comment:
For the exception the membertype will always be user. Did you want to have this at the same indention level as the try/except?

Reply:
Actually, at this level it would always be 'group'. I moved it to the outer try/except, this way it would be either group or user.